main/tests/server/src/com/google/refine/grel/GrelTests.java

```diff
@@ -165,18 +188,30 @@ public void testPI() throws ParsingException {
     @Test
     public void testString() throws ParsingException {
         String tests[][] = {
-                { "1", "1" },
-                { "1 + 1", "2" },
-                { "1 + 1 + 1", "3" },
-                { "1-1-1", "-1" },
-                { "1-2-3", "-4" },
-                { "1-(2-3)", "2" },
-                { "2*3", "6" },
-                { "3%2", "1" },
-                { "3/2", "1" },
-                { "3.0/2", "1.5" },
-                { "1", "1" },
-                { "0/0", "NaN" },
+                { "'a' + 'b'", "ab" },
+                // TODO: automated fuzzing of all operators for incompatible operand types
+                { "'1/1/1900'.toDate() + 1", null },
+                { "'1/1/1900'.toDate() + '1/1/1800'.toDate()", null },
+                { "'1/1/1900'.toDate() > '1/1/1800'.toDate()", "true" },
+                { "'1/1/1900'.toDate() >= '1/1/1800'.toDate()", "true" },
+                { "'1/1/1900'.toDate() < '1/1/1800'.toDate()", "false" },
+                { "'1/1/1900'.toDate() <= '1/1/1800'.toDate()", "false" },
+                { "'1/1/1900'.toDate() != '1/1/1800'.toDate()", "true" },
+                { "'1/1/1900'.toDate() == '1/1/1800'.toDate()", "false" },
+                { "'1/1/1900'.toDate() == '1/1/1900'.toDate()", "true" },
+                { "'1/1/1900'.toDate() >= '1/1/1900'.toDate()", "true" },
+                { "'1/1/1900'.toDate() <= '1/1/1900'.toDate()", "true" },
+                { "'1/1/1900'.toDate() + ' foo'", "1900-01-01T00:00Z foo" },
+                { "1 + ' foo'", "1 foo" },
+                { "1.0 + ' foo'", "1.0 foo" },
+                { "2 * 3.0 + ' foo'", "6.0 foo" },
+                { "'a' > 'b'", "false" },
+                { "'a' < 'b'", "true" },
+                { "'a' == 'a'", "true" },
+                { "'a' == 'b'", "false" },
+                { "'a' != 'b'", "true" },
+                { "'E\u0301' == 'É'", "true" }, // combining accent equivalent to single character form
```

Comment:
@tfmorris So how does a user control the CANONICAL_DECOMPOSITION itself?  Like for instance, if a power user does not want to accept accent equivalency and only wants to accept String literal checking for equality?  Do we need to add some other parameter to control their preference for that?

Reply:
Obviously operators can't take additional parameters. I chose this as the default because I think it meets the principal of least surprise, but it's not out of the question to introduce a preference setting for the "strength" of string equality to ignore composition, accents, or even case. If you think this is an important feature, please create an issue for it.

Also, if you have a strong argument for why the default should be different, I'm happy to discuss. In my experience, the majority of users want things that look the same to be considered equal.